pymathics/natlang/textual_analysis.py

```diff
@@ -0,0 +1,430 @@
+
+    <dl>
+      <dt>'SpellingCorrectionList[$word$]'
+      <dd>returns a list of suggestions for spelling corrected versions of $word$.
+    </dl>
+
+    Results may differ depending on which dictionaries can be found by enchant.
+
+    >> SpellingCorrectionList["hipopotamus"]
+     = {hippopotamus...}
+    """
+
+    options = {
+        "Language": '"English"',
+    }
+
+    messages = {
+        "lang": "SpellingCorrectionList does not support `1` as a language.",
+    }
+
+    _languages = {
+        "English": "en_US",  # en_GB, en_AU
+        "German": "de_DE",
+        "French": "fr_FR",
+    }
+
+    _dictionaries = {}
+
+    summary_text = "Look for spelling correction candidates of a word"
+
+    def eval(
+        self, word: String, evaluation: Evaluation, options: dict
+    ) -> Optional[ListExpression]:
+        "SpellingCorrectionList[word_String, OptionsPattern[SpellingCorrectionList]]"
+
+        language_name = self.get_option(options, "Language", evaluation)
+        if not isinstance(language_name, String):
+            return
+        language_code = SpellingCorrectionList._languages.get(language_name.value, None)
+        if not language_code:
+            evaluation.message("SpellingCorrectionList", "lang", language_name)
+            return
+
+        d = SpellingCorrectionList._dictionaries.get(language_code, None)
+        if not d:
+            d = enchant.Dict(language_code)
+            SpellingCorrectionList._dictionaries[language_code] = d
+
+        py_word = word.value
+
+        if d.check(py_word):
+            return ListExpression(word)
+        else:
+            return ListExpression(*(String(word) for word in d.suggest(py_word)))
+
+
+class WordCount(_SpacyBuiltin):
+    """
+    <url>:WMA link:
+    https://reference.wolfram.com/language/ref/WordCount.html</url>
+
+    <dl>
+      <dt>'WordCount[$string$]'
+      <dd>returns the number of words in $string$.
+    </dl>
+
+    >> WordCount["A long time ago"]
+     = 4
+    """
+
+    summary_text = "Count the words in a text"
+
+    def eval(self, text, evaluation: Evaluation, options: dict):
+        "WordCount[text_String, OptionsPattern[WordCount]]"
+        doc = self._nlp(text.value, evaluation, options)
+        if doc:
+            punctuation = spacy.parts_of_speech.PUNCT
+            return Integer(sum(1 for word in doc if word.pos != punctuation))
+
+
+class WordFrequency(_SpacyBuiltin):
+    """
+    <url>:WMA link:
+    https://reference.wolfram.com/language/ref/WordFrequency.html</url>
+
+    <dl>
+      <dt>'WordFrequency[$text$, $word$]'
+      <dd>returns the relative frequency of $word$ in $text$.
+    </dl>
+
+    $word$ may also specify multiple words using $a$ | $b$ | ...
+
+    ## Problem with import for certain characters in the text.
+    ## >> text = Import["ExampleData/EinsteinSzilLetter.txt"];
+    >> text = "I have a dairy cow, it's not just any cow. \
```

Comment:
Wrapping does not render right for "\". Look at the Dango renderign here.

Reply:
It is a pity. Now I removed the extra spaces just by eliminating the spaces from the left indent. I also noticed that in Django, linebreaks are ignored when strings are shown.